MediaBrowser.Providers/Manager/ProviderManager.cs

```diff
@@ -276,19 +273,19 @@ public async Task<IEnumerable<RemoteImageInfo>> GetAvailableRemoteImages(BaseIte
                     result = result.Where(i => i.Type == type.Value);
                 }
 
-                if (!includeAllLanguages && preferredLanguages.Count > 0)
+                if (!includeAllLanguages && hasPreferredLanguage)
                 {
                     // Filter out languages that do not match the preferred languages.
                     //
                     // TODO: should exception case of "en" (English) eventually be removed?
                     result = result.Where(i => string.IsNullOrEmpty(i.Language) ||
-                                               preferredLanguages.Contains(i.Language, StringComparer.OrdinalIgnoreCase) ||
+                                               string.Equals(preferredLanguage, i.Language, StringComparison.OrdinalIgnoreCase) ||
                                                string.Equals(i.Language, "en", StringComparison.OrdinalIgnoreCase));
                 }
-                else if (preferredLanguages.Count > 0)
+                else if (hasPreferredLanguage)
                 {
                     // Order the images by preferred languages first.
-                    result = result.OrderBy(i => !string.IsNullOrEmpty(i.Language) && preferredLanguages.Contains(i.Language, StringComparer.OrdinalIgnoreCase) ? 0 : 1);
+                    result = result.OrderByLanguageDescending(preferredLanguage);
```

Comment:
You should remove the call to OrderByLanguageDescending in the tmdb providers and remove the `else if (hasPreferredLanguage)`. The orderby should probably be executed no matter what.